paderbox/utils/random_utils.py

```diff
@@ -25,73 +27,290 @@ def _force_correct_shape(f):
+    ((2, 3), dtype('float64'))
+    >>> x = randn(2, 3, dtype=np.complex128)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('complex128'))
+    """
+    return np.random.randn(*shape)
+
+
+def normal(*shape, loc=0., scale=1., dtype=np.complex128):
+    """
 
-    def _randn(data_type_local):
-        return np.random.randn(*shape).astype(data_type_local)
+    Args:
+        *shape:
+        loc:
+        scale:
+        dtype:
 
-    if dtype in (np.float32, np.float64):
-        return _randn(dtype)
-    elif dtype is np.complex64:
-        return _randn(np.float32) + 1j * _randn(np.float32)
-    elif dtype is np.complex128:
-        return _randn(np.float64) + 1j * _randn(np.float64)
+    Returns:
 
+    """
+    return scale * randn(*shape, dtype=dtype) + loc
+
+
+@_force_correct_shape
+@_add_kwarg_dtype(default=np.float64)
+def truncated_normal(*shape, loc=0., scale=1., truncation=3.):
+    """
+
+    Args:
+        *shape:
+        loc:
+        scale:
+        truncation:
+        dtype:
 
-def normal(*shape, dtype=np.complex128):
-    return randn(*shape, dtype=dtype)
+    Returns:
+
+    >>> x = truncated_normal()
+    >>> x.ndim
+    0
+    >>> x = truncated_normal(2, 3)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('float64'))
+    >>> x = truncated_normal(2, 3, dtype=np.complex128)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('complex128'))
+    """
+    return (
+        truncnorm(-truncation / scale, truncation / scale, loc, scale).rvs(shape)
+    )
 
 
 @_force_correct_shape
-def hermitian(*shape, data_type=np.complex128):
+@_add_kwarg_dtype(default=np.float64)
+def log_truncated_normal(*shape, loc=0., scale=.5, truncation=3.):
+    """
+
+    Args:
+        *shape:
+        loc:
+        scale:
+        truncation:
+        dtype:
+
+    Returns:
+
+    >>> x = log_truncated_normal()
+    >>> x.ndim
+    0
+    >>> x = log_truncated_normal(2, 3)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('float64'))
+    >>> x = log_truncated_normal(2, 3, dtype=np.complex128)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('complex128'))
+    """
+    return np.exp(truncnorm(-truncation / scale, truncation / scale, loc, scale).rvs(shape))
+
+
+@_force_correct_shape
+@_add_kwarg_dtype(default=np.float64)
+def truncated_exponential(*shape, loc=0., scale=1., truncation=3.):
+    """
+
+    Args:
+        *shape:
+        loc:
+        scale:
+        truncation:
+
+    Returns:
+
+    >>> x = truncated_exponential()
+    >>> x.ndim
+    0
+    >>> x = truncated_exponential(2, 3)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('float64'))
+    >>> x = truncated_exponential(2, 3, dtype=np.complex128)
+    >>> x.shape, x.dtype
+    ((2, 3), dtype('complex128'))
+    """
+    return truncexpon(truncation / scale, loc, scale).rvs(shape)
+
+
+@_force_correct_shape
+def hermitian(*shape, dtype=np.complex128):
     """ Assures a random positive-semidefinite hermitian matrix.
 
     :param shape:
-    :param data_type:
+    :param dtype:
```

Comment:
Same as above (and for the following function): Can make sure that all docstrings use the same style?